packages/jest-runtime/src/writeFileAtomic.js

```diff
@@ -0,0 +1,219 @@
+/* eslint-disable */
+
+// THIS FILE IS COPIED FROM https://github.com/npm/write-file-atomic/blob/1597785cb9e1152056d905a357b3891b3295506d/index.js
+
+// Forked in order to support running in worker threads, where process.pid is shared with the main thread
+
+'use strict'
+module.exports = writeFile
+module.exports.sync = writeFileSync
+module.exports._getTmpname = getTmpname // for testing
+module.exports._cleanupOnExit = cleanupOnExit
+
+var fs = require('graceful-fs')
+var MurmurHash3 = require('imurmurhash')
+var onExit = require('signal-exit')
+var path = require('path')
+var activeFiles = {}
+
+// This is the addition over the original code
+var id = (function getId() {
+  try {
+    const workerThreads = require('worker_threads')
+    if (workerThreads.isMainThread) {
+      return process.pid
+    }
+
+    return workerThreads.threadId || process.pid
```

Comment:
We can probably assume that it will either end up with `threadId` for all workers or with `pid` for all? Otherwise, `return -workerThreads.threadId || process.pid` will make them unique :D

Reply:
I have no idea

Reply:
it'll only be `0` in one though, so should still be unique! 😀

Reply:
Not if the pid is equal to the thread id of another one :P